### PR TITLE
add public key support from a file

### PR DESCRIPTION
**Pull request by tedder**
- refactor the commands to use string.format. This allows multiple runs (needed for 'discovered' information).
- add key_type and key_file. Currently key_id goes with key_type=private and key_file with key_type=public, but that will change as a private key import from file is added. Might also take the key_id as verification of the file contents too.
- add a simple debug/logger. I've found this handy in other modules.

### CODE REVIEW

**Comment by tedder** [comment]
@conorsch hey- FYI on this, per your request.

**Comment by conorsch** [comment]
@tedder Nice! Going to pull this into a local site config and take it for a spin.

File: gpg_import.py [hunk 1]
```diff
@@ -17,10 +18,23 @@
...
+      - Filename of key to be imported. Must be on remote machine, not local. Only applicable to public keys (for now). Either key_file or key_id is required.
+    required: false
     default: null
 
+  key_type:
+    description:
+      - What type of key to import.
+    required: true
+    choices: [ "private", "public" ]
+    default: "private"
```
Comment (conorsch) [inline gpg_import.py:36]: Defaulting to private key imports seems unintuitive—I'd wager most gpg-related import tasks are for public keys, e.g. for verifying signatures.

Comment (tedder) [reply gpg_import.py:36]: I'm always conservative, private was the default case here. Perhaps there should be no default here?

**Pull request merged**